src/charm.py

```diff
@@ -132,19 +141,9 @@ def _on_start(self, event: StartEvent):
         # Remove some config entries when cluster bootstrapped
         self._cleanup_conf_if_bootstrapped(nodes)
 
-        # start opensearch
-        if not self._start_opensearch():
-            event.defer()
-            return
-
-        # initialize the security index if the admin certs are written on disk
-        if self.unit.is_leader():
-            if self.app_peers_data.get("security_index_initialised") is None:
-                admin_secrets = self.secrets.get_object(Scope.APP, CertType.APP_ADMIN.val)
-                self._initialize_security_index(admin_secrets)
-                self.app_peers_data["security_index_initialised"] = "True"
-
-            self.app_peers_data["leader_ip"] = self.unit_ip
+        # request the start of opensearch
+        self.unit.status = WaitingStatus(RequestUnitServiceOps.format("start"))
+        self.on[self.service_manager.name].acquire_lock.emit(callback_override="_start_opensearch")
```

Comment:
I think it's a great suggestion! I technically don't _need_ it - I just added it for clarity in the code, so that when I read it, I can quickly see the target callback to the current action (really just for my own readability)